### PR TITLE
types: added response to FastifySchema

### PR DESCRIPTION
**Pull request by MichielDeMey**
The `FastifySchema` was missing the optional response type definition

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: types/schema.d.ts [hunk 1]
```diff
@@ -9,6 +9,10 @@ export interface FastifySchema {
   querystring?: unknown;
   params?: unknown;
   headers?: unknown;
+  response?: {
+    [code: number]: unknown;
```
Comment (Ethan-Arrowood) [inline types/schema.d.ts]: please make this `[code: number | string]: unknown`

Comment (Ethan-Arrowood) [reply types/schema.d.ts]: Just realizing that this syntax is invalid. Oops my mistake!

Comment (Ethan-Arrowood) [reply types/schema.d.ts]: I'd rather this be `unknown` like the rest of the properties. Then encourage the user to specify it themselves using overloading or generics.

Comment (MichielDeMey) [reply types/schema.d.ts]: It does give more flexibility in terms of typing your schema. I'll change it. 👍

**Comment by Ethan-Arrowood** [comment]
I'm not sure why the CI is failing. I tried restarting it and it keeps failing in the initialization step - maybe this is GitHub actions problem. I'll revisit later and try it again; obviously, this change isn't causing this issue so it'll get merged asap anyways. Thanks for the contribution!

**Comment by MichielDeMey** [comment]
Can this be merged without passing tests though? 😅

**Comment by Ethan-Arrowood** [comment]
I would say yes considering the change but I want to bring the CI failure to @fastify/core attention so we can figure out whats going wrong

**Comment by Ethan-Arrowood** [comment]
ping @MichielDeMey are you able to create a test for this type? No worries if you can't, i'm happy to do it and we can get this merged 😄

**Comment by MichielDeMey** [comment]
@Ethan-Arrowood  Sorry, got lost on my radar. 🙂
I don't have a lot of experience with testing TypeScript types, so it might be faster if you add the tests for this. 👍

**Comment by mcollina** [comment]
Just add a few lines using this new type to https://github.com/fastify/fastify/blob/master/test/types/index.ts.

**Comment by Ethan-Arrowood** [comment]
@mcollina this is targeting _next_ 😉 

No worries @MichielDeMey testing is very similar to unit - testing. Let me give you some pointers

**Comment by Ethan-Arrowood** [comment]
I'll add this to my todo to finish up adding the tests.

**Comment by delvedor** [comment]
Please rebase this pr against the new `master` branch, see https://github.com/fastify/fastify/issues/2056#issuecomment-612916945.

**Comment by Ethan-Arrowood** [comment]
Going to finish up this PR right now; rebasing and adding tests.

**Comment by Ethan-Arrowood** [comment]
@MichielDeMey can you enable me to push to your branch? It seems to be disabled

**Comment by MichielDeMey** [comment]
@Ethan-Arrowood Sorry, should be working now. 🙂

**Comment by Ethan-Arrowood** [comment]
I still can't seem to push. Can you merge the changes i've made here? https://github.com/Ethan-Arrowood/fastify/tree/michieldemey-patch-2

**Comment by MichielDeMey** [comment]
@Ethan-Arrowood I merged your changes into my branch 👍

**Comment by mcollina** [comment]
Would you mind to add a test for this type? Thanks!

**Comment by MichielDeMey** [comment]
I'll try, but I'm not familiar with testing TypeScript types.
I'm assuming I can extend https://github.com/fastify/fastify/blob/master/test/types/route.test-d.ts to also include `FastifySchema` type testing?

**Comment by mcollina** [comment]
Yes, exactly.

**Comment by SystemDisc** [comment]
Until this is merged and a new release-candidate for v3 comes out, I've resorted to adding the following to my `server.ts`:

```typescript
declare module 'fastify' {
  interface FastifySchema {
    response?: unknown;
  }
}
```

**Comment by mcollina** [comment]
@SystemDisc would you like to send a fresh PR that include this change as well as unit tests?

**Comment by SystemDisc** [comment]
This PR includes the FastifySchema change. If I have time later, I can put up a PR for the other change.

**Comment by mcollina** [comment]
this is missing the test

**Comment by kibertoad** [comment]
Probably this is made redundant by #2295

**Comment by MichielDeMey** [comment]
Correct, looks like this made it into #2295. 👍
Closing this.

**Comment by zmeyc** [comment]
@SystemDisc @MichielDeMey @kibertoad My PR didn't include fix for addSchema, I now added it.

**Pull request closed without merging**